### PR TITLE
doctor/ethtool, ipn/ipnlocal: add ethtool bugreport check

### PR DESCRIPTION
**Pull request by andrew-d**
Updates #11137

Change-Id: Idbe862d80e428adb044249c47d9096b87f29d5d8

### CODE REVIEW

**Comment by andrew-d** [comment]
Example output in `tailscale bugreport --diagnose`:

```
diag: ethtool: lo: features: [tx-checksum-sctp tx-sctp-segmentation tx-udp-segmentation highdma tx-tcp6-segmentation rx-gro rx-checksum tx-gso-list tx-lockless tx-checksum-ip-generic tx-tcp-mangleid-segmentation tx-tcp-ecn-segmentation tx-tcp-segmentation vlan-challenged netns-local tx-generic-segmentation tx-scatter-gather-fraglist loopback tx-scatter-gather]
diag: ethtool: ens5: features: [rx-hashing tx-checksum-ipv4 rx-gro tx-generic-segmentation highdma tx-scatter-gather]
diag: ethtool: ens5: warning: pps_allowance_exceeded = 99 > 0
diag: ethtool: ens5: warning: bw_in_allowance_exceeded = 1865 > 0
diag: ethtool: tailscale0: features: [tx-generic-segmentation tx-lockless tx-scatter-gather-fraglist tx-tcp6-segmentation tx-vlan-hw-insert tx-tcp-segmentation rx-gro tx-scatter-gather tx-vlan-stag-hw-insert tx-checksum-ip-generic tx-udp-segmentation]
```

File: doctor/ethtool/ethtool_linux.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+	if err != nil {
+		logf("could not create ethtool: %v", err)
+		return nil
+	}
+	defer et.Close()
+
+	interfaces.ForeachInterface(func(iface interfaces.Interface, _ []netip.Prefix) {
+		ilogf := logger.WithPrefix(logf, iface.Name+": ")
+		features, err := et.Features(iface.Name)
+		if err == nil {
```
Comment (raggi) [inline doctor/ethtool/ethtool_linux.go:26]: maybe wanna log err != nil to the logclient so if it comes up in the wild we can dig it up and try to address it?

**Review by raggi — APPROVED** [review]

File: doctor/ethtool/ethtool_linux.go (hunk 1 above)
Comment (andrew-d) [reply doctor/ethtool/ethtool_linux.go:27]: Good call; added both here and in the `Stats` call

File: doctor/ethtool/ethtool_linux.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+	"conntrack_allowance_exceeded",
+	"linklocal_allowance_exceeded",
+	"pps_allowance_exceeded",
+})
+
+// Stats that should be printed if zero
+var zeroStats = set.SetOf([]string{
+	// AWS ENA driver statistics; see:
+	//    https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/monitoring-network-performance-ena.html
+	"conntrack_allowance_available",
```
Comment (sailorfrag) [inline doctor/ethtool/ethtool_linux.go:66]: Would it be useful to also print that if the value is "small" but not zero?

Comment (andrew-d) [reply doctor/ethtool/ethtool_linux.go:66]: Maybe, but if it ever hit zero (even if it's currently "small"), then the `conntrack_allowance_exceeded` counter will be non-zero and should be logged as well. Might leave this as-is for now, but I want to integrate these statistics into a more general health concept that we can report to the controlplane, at which point checking for small-but-nonzero is definitely worth doing.

**Pull request merged**